src/NHapi.Base/Model/Varies.cs

```diff
@@ -170,8 +170,28 @@ public static void fixOBX5(ISegment segment, IModelClassFactory factory)
 					{
 						UseDTInsteadOfDTMForEarlierVersionsOfHL7(segment, obx2);
 
-						Type c = factory.GetTypeClass(obx2.Value, segment.Message.Version);
-						v.Data = (IType)c.GetConstructor(new[] { typeof(IMessage), typeof(String) }).Invoke(new Object[] { v.Message, v.Description });
+						Type type = factory.GetTypeClass(obx2.Value, segment.Message.Version);
+
+						if(type == null)
+						{
+							var obx1 = (IPrimitive) segment.GetField(1, 0);

```

Comment:
could `obx1` be null? if so, you might get a null ref when accessing `.Value` below?

Reply:
You are right it could be null, but it is caught further down so should be ok - as it happens this code is ported/copied/borrowed from [hapi](https://github.com/hapifhir/hapi-hl7v2/blob/efca803e58aea1705ba6b28390ad24c4c3b1d3cb/hapi-base/src/main/java/ca/uhn/hl7v2/parser/FixFieldDataType.java#L220) so at least it's got parity with that 😄